autogen/agentchat/contrib/gpt_assistant_agent.py

```diff
@@ -21,33 +21,29 @@ class GPTAssistantAgent(ConversableAgent):
     def __init__(
         self,
         name="GPT Assistant",
-        instructions: Optional[str] = "You are a helpful GPT Assistant.",
+        instructions: Optional[str] = None,
         llm_config: Optional[Union[Dict, bool]] = None,
+        overwrite_instructions: bool = False,
     ):
         """
         Args:
             name (str): name of the agent.
             instructions (str): instructions for the OpenAI assistant configuration.
+            the system message of the agent will be set to the instructions and used in the assistant run.
+            (irrespective of the overwrite_instructions flag)
             llm_config (dict or False): llm inference configuration.
                 - model: Model to use for the assistant (gpt-4-1106-preview, gpt-3.5-turbo-1106).
                 - check_every_ms: check thread run status interval
                 - tools: Give Assistants access to OpenAI-hosted tools like Code Interpreter and Knowledge Retrieval,
                         or build your own tools using Function calling. ref https://platform.openai.com/docs/assistants/tools
                 - file_ids: files used by retrieval in run
+            overwrite_instructions (bool): whether to overwrite the instructions of an existing assistant.
         """
-        super().__init__(
-            name=name,
-            system_message=instructions,
-            human_input_mode="NEVER",
-            llm_config=llm_config,
-        )
-
         # Use AutoGen OpenAIWrapper to create a client
-        oai_wrapper = OpenAIWrapper(**self.llm_config)
+        oai_wrapper = OpenAIWrapper(**llm_config)
         if len(oai_wrapper._clients) > 1:
             logger.warning("GPT Assistant only supports one OpenAI client. Using the first client in the list.")
         self._openai_client = oai_wrapper._clients[0]
-
         openai_assistant_id = llm_config.get("assistant_id", None)
         if openai_assistant_id is None:
```

Comment:
Under this condition, we may have the case where `instructions` is not set (default to None). In this case should we  use the default instructions of GPT assistant?

Reply:
improved the doc string. Please check and resolve if it looks good.